### ISSUE TITLE
A question about beam search

### ISSUE THREAD

**Issue by Marcovaldong** [ZhengkunTian/rnn-transducer#1]
@ZhengkunTian Thanks for your code. I have readen your [blog](https://zhuanlan.zhihu.com/p/62050647) about the rnn-transducer. 

I want to know how did you implement the procedure of beam searching. I implemented this procedure using PyTorch by referring [HawkAaron's implementation](https://github.com/HawkAaron/RNN-Transducer/blob/graves2013/model.py#L100), but I found the beam search is too slow to use. So I wonder to know your implementation and your speed. 

I'll be very appreciate if you can answer me.

**Comment by ZhengkunTian** [comment]
I’m so sorry to answer it so late. It's so hard to achieve a batch beam search method. In my current implementation, I utilize a greedy search or decode sentences one by one with beam search. In my experiments, I find that beam search just bring a little improvement. Therefore, I don't achieve an efficient beam search code. Thanks for your attention.

**Comment by Marcovaldong** [comment]
@ZhengkunTian Thanks for your reply and congratulations for your papaers accepted by interspeech2019. The focus of my problem is not how to implement the beam search on a mini-batch, but how to implement it on a single sample and how its speed and performance. 

On the other hand, I tried to implement shallow fusion in greedy decoding, can we do that?  I found my implementation of shallow fusion with PyTorch is invalid. However I added my language model which was implemented by MxNet in shallow fusion with greedy decoding, there is an improvement of 0.2%. So I want to know the performance of shallow fusion in your experiments.
